Ink playground milestone 1

radhe-zeeve opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, an invoice must be submitted and the payment will be transferred to the Polkadot/fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1731

PieWol commented on deliveries/ink-playground-milestone-1.md:
```diff
@@ -0,0 +1,31 @@
+| 0.b. | Documentation | https://github.com/Zeeve-App/ink-playground/blob/version_play/README.md | We will provide both inline documentation of the code and a basic tutorial that explains how a user can open and save a file |
+| 1.  | On demand version specific compilation | https://github.com/paritytech/ink-playground/pull/815 | We will add the ability to build the compiler environment if a compiler environment is not present with a specific version of Ink! |
+| 2. | Dependency Version support | https://github.com/paritytech/ink-playground/pull/815, https://github.com/paritytech/ink-playground/pull/817 | We will upgrade and maintain Ink! and cargo-contract dependencies versions up to date and have provision to have older version support |
+| 3. | Dependency Version support | https://github.com/paritytech/ink-playground/pull/815 | We will add the ability to add Playground's Ink! version support to latest version as soon as new version of Ink! is released without manual intervention |
```
```suggestion
| 3. | Ink! Upgrades | https://github.com/paritytech/ink-playground/pull/815 | We will add the ability to add Playground's Ink! version support to latest version as soon as new version of Ink! is released without manual intervention |
```

PieWol commented:
Hey @radhe-zeeve thanks for the delivery!
Before I test the functionality and build it locally I would like to know if the branch you linked for documentation (version play) is the right branch to use to test all functionalities you added? Thanks for also linking the individual PR's.

radhe-zeeve commented:
@PieWol Yes, most active 
of the features mentioned are built in `Zeeve-App:version_play` branch and only version update through CI was managed in `Zeeve-App:new_version`

PieWol commented:
@radhe-zeeve I'm having troubles following the readme both on ubuntu and macOS. Can you please make sure that the current way of setting this project up on a fresh machine works according to the readme? As in pull the repo freshly somewhere and start the build process from scratch. 

On my arm mac it's failing due to 
```Compiling chalk-solve v0.83.0
error: failed to build archive: section too large

ink-playground/target/wasm32-unknown-unknown/release/build/psm-2f66b15b10c4600a/out/libpsm_s.a the table of contents is empty (no object file members in the library define global symbols)

error: could not compile `psm` due to previous error
```

and on ubuntu 
```root@ip-172-31-24-165:~/ink-playground# make install
yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
make: *** [Makefile:216: ts-install] Error 1
```

Thank you.

radhe-zeeve commented:
@PieWol  On Mac it might not work due to ARM architecture.
As for Ubuntu 22.04 on x86_64 arch (2 Core 8 GB). It works
```sh
sudo apt update
sudo apt install make
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.7/install.sh | bash
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
nvm install 18
npm i -g yarn
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
source "$HOME/.cargo/env"
sudo apt install gcc
sudo apt install libssl-dev pkg-config
https://github.com/Zeeve-App/ink-playground.git
cd ink-playground
git checkout version_play
make install
make generate
make build
```

PieWol commented:
Hey @radhe-zeeve thanks for providing the instructions you expect to work. I have followed it as you said on ubuntu 22.04 and I'm still running into issues. 

1. on `make generate` the wasm optimization fails.
2. `make build` fails afterwards. 

I have included all relevant information to this in this [pastebin ](https://pastebin.com/0txeeDBe). Please check it out and get back to me on how we can overcome this issue. Thank you.

radhe-zeeve commented:
@PieWol 
Please refer to the recording of the setup done on fresh Ubuntu instance: https://asciinema.org/a/636353
we can have a call if still not resolved.

PieWol commented:
Hey @radhe-zeeve , 
in the documentation deliverable you stated that

> We will provide ... a basic tutorial that explains how a user can open and save a file

I can't seem to find that tutorial in the linked readme file. Where can I find this user focussed tutorial?

PieWol commented:
Hey @radhe-zeeve ,
just checking in on you to see if there have been any updates you might be able to share.

radhe-zeeve commented:
> Hey @radhe-zeeve , just checking in on you to see if there have been any updates you might be able to share.

Will be provide the doc and answers within couple of days

radhe-zeeve commented:
@PieWol 

- Updated documentation for testing check "Run Production service without dockerized image", "General usage",  "Issues with running on remote server without HTTPS support" sections
- > For the automated ink! version upgrade you wrote this [github action](https://github.com/paritytech/ink-playground/actions/workflows/ink_version_update.yml), right? Why does it have no runs yet? Shouldn't it run everyday 6am GMT?
  - Answer: Since the repository is archived all the actions are disabled, I have pulled and changed the time to schedule the action and to execute it (https://github.com/Zeeve-App/ink-playground/actions/runs/7897895698/job/21554248329), the action did not create new PR because there was already a PR present in the repo with same name and open status (https://github.com/Zeeve-App/ink-playground/pull/7). 
-  `We will add the ability to build the compiler environment if a compiler environment is not present with a specific version of Ink!` this was part of the version update where if the version is not present it will update the contract dependencies to new Ink version and add it to `versions.json`, then after testing the PR we can merge and also add docker file (in `docker` directory) with new version to have new Ink version support.

PieWol commented:
Hey @radhe-zeeve thanks for your latest update!

I see how your github action updates the versions available for the users of the ink!-playground. Thats good and seems to be working as expected given the link to the upgrade PR and the github action output.

Thanks also for clarifying how this last deliverable that I was unsure about actually works. Now the way I understand is that you are preparing the ink!-playground code so that it will be able to interact with a dockerized compiler for the latest version once you have added the dockerized compiler as a dependancy for the repo. Is that correct? So "adding the possibility" translates to "preparing dependencies for a to be released dockerized compiler for the latest version"?

I would appreciate it if you could confirm my understand a final time. Thanks for your help so far!

keeganquigley commented:
Thanks @radhe-zeeve @PieWol has accepted your milestone, here is his [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/ink_playground_1_PieWol.md). I will hereby merge the delivery. In the meantime feel free to answer his remaining question above.

semuelle commented:
@radhe-zeeve, the postal address in your invoices doesn't match the one in your application. Could you modify one to match the other? Thanks!

radhe-zeeve commented:
@semuelle sure, will update the address, upload and notify here.

radhe-zeeve commented:
@semuelle  the office address has been changed, the address in the invoice is the latest and canonical address. Do we have to change the application for this ?

semuelle commented:
> Do we have to change the application for this ?

Yes, please. Just submit a pull request with the changed address. Should be quickly approved.

radhe-zeeve commented:
@semuelle PR got merged.

semuelle commented:
@radhe-zeeve, I'm afraid this won't be settled today, so most likely next Friday.

RouvenP commented:
hi @radhe-zeeve we sent the payment today
